src/mongocxx/result/bulk_write.hpp

```diff
@@ -33,6 +34,8 @@ namespace result {
 class MONGOCXX_API bulk_write {
 
    public:
+    using id_map = std::map<std::size_t, bsoncxx::document::element>;
```

Comment:
I think ordering is nice, since it makes it easier to iterate through the upserted IDs in the order they were created.